imxrt-hal/src/srtc.rs

```diff
@@ -117,27 +118,29 @@ fn is_enabled(snvs: &Instance) -> bool {
     ral::read_reg!(ral::snvs, snvs, LPCR, SRTC_ENV) != 0
 }
 
-/// Converts the number of milliseconds that have occurred since a second into clock ticks
+/// Converts the number of microseconds that have occurred since a second into clock ticks
 /// (1/32768 of a second).
 ///
-/// For example: for the time 2020-10-05 01:39:56.505, `millis` is `505`.
-pub fn subsec_millis_to_ticks(millis: u16) -> u16 {
-    // 999 is the max valid value outside of a leap second; clamp
-    let millis = millis.min(999);
-    let millis = millis as f32 / 1000.0;
-    let ticks = millis * 32768.0;
+/// For example: for the time `2020-10-05 01:39:56.505`, `micros` is `505000`, and this gives
+/// `16547` ticks.
+pub fn subsec_micros_to_ticks(micros: u32) -> u16 {
+    const RATIO: f32 = 0.032768; // 32768 / 1_000_000, each microsecond is this many ticks
+    let micros = micros.min(999_999);
+    // 999999 is the max valid value outside of a leap second; clamp
+    let ticks = micros as f32 * RATIO;
```

Comment:
So doing integer math usually what you want to do is scale to some more precise unit first, do math, then scale back to the less precise unit. So for example.... 1_000_000_000/32_768 gets you the number of nanoseconds per tick (30517) as an integer. Convert the microseconds to nano seconds (multiply by 1000), divide by nanoseconds per tick, get the number of ticks.

So in real terms.... 
``` rust
const NANOS_PER_TICK: u32 = 30517;
fn microseconds_to_ticks(micros: u32) -> u16 { 
  let nanos = micros *1000;
  nanos/NANOS_PER_TICK
}
```

Reply:
I'm not sure this function needs to be public, if it is lets avoid the micros.min() and call it micros_to_ticks

Reply:
Thanks!

I've changed the code to use quarter-nanoseconds per tick, which is slightly more accurate than just nanoseconds but still fits within 32-bit unsigned math.

`micros_to_ticks` needs to be public as it is not used internally, it's for users to convert sub-second time into clock ticks so they can more easily set the time.

I've removed the `min` call and documented the max valid value instead.